src/flows/recommendation_api/user_impressions_flow.py

```diff
@@ -1,49 +1,84 @@
-from prefect import task, Flow, Parameter, unmapped
-from prefect.tasks.aws.s3 import S3List
+from prefect import task, Flow, Parameter, context
+from api_clients.pocket_snowflake_query import PocketSnowflakeQuery
 import pandas as pd
+import datetime
 import boto3
 from sagemaker.feature_store.feature_group import FeatureGroup
 from sagemaker.session import Session
+from prefect.schedules import IntervalSchedule
 
+from utils import config
 from utils.flow import get_flow_name
 
 FLOW_NAME = get_flow_name(__file__)
 
-
-@task
-def extract_parquet_as_df(bucket:str, parquet_file: str):
-    parquet_file = f"s3://{bucket}/{parquet_file}"
-    return pd.read_parquet(parquet_file)
+BASE_QUERY = """
+WITH 
+prep as (
+  SELECT
+    a.HASHED_USER_ID,
+    a.CONTENT_ID,
+    c.RESOLVED_ID,
+    MAX(a.IMPRESSION_AGE) as max_impression_age,
+    SUM(a.IMPRESSION_COUNT) as total_imprs,
+    MAX(a.HAPPENED_AT_DAY) as HAPPENED_AT
+  FROM ANALYTICS.DBT.FIRST_IMPRESSED_AGE as a
+  JOIN ANALYTICS.DBT.CONTENT AS c
```

Comment:
Is it possible to use only `content_id` instead of `resolved_id`? This join seems unnecessary.
If we are going to need `resolved_id`s then we should consider putting those in the models (example: `FIRST_IMPRESSED_AGE`) we use and avoid the joins

Reply:
this choice is related to the recommendation-api which doesn't currently know about content_id since it is older than that

Reply:
@mmiermans has been working on this. We may be closer than you think.